src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -137,9 +145,40 @@ static Controller fromOptions(Environment env,
         int scalingParameter = options.containsKey(STARTING_SCALING_PARAMETER) ? Integer.parseInt(options.get(STARTING_SCALING_PARAMETER)) : DEFAULT_STARTING_SCALING_PARAMETER;
         int[] scalingParameters = new int[32];
         int[] previousScalingParameters = new int[32];
-        //set the scaling parameter for each level to the starting scaling parameter or default scaling parameter
-        Arrays.fill(scalingParameters, scalingParameter);
-        Arrays.fill(previousScalingParameters, scalingParameter);
+        //check to see if we have scaling parameters stored and read those if able, otherwise use starting values
+        File f = new File(SCALING_PARAMETER_FILE_PATH, SCALING_PARAMETER_FILE_NAME);
+        if (f.length() != 0 && f.isReadable())
+        {
+            try
+            {
+                BufferedReader br = new BufferedReader(new FileReader(f));
+                String scalingParametersString = br.readLine();
+                String[] scalingParametersStrings = scalingParametersString.split(",");
+                logger.debug(String.format("Reading stored scaling parameters from %s: %s", f.toPath().toString(), Arrays.toString(scalingParametersStrings)));
+                for (int i = 0; i < scalingParameters.length; i++)
+                {
+                    //if the file does not have enough entries, use the last entry for the rest of the levels
+                    if (i < scalingParametersStrings.length)
+                        scalingParameters[i] = Integer.parseInt(scalingParametersStrings[i]);
+                    else
+                        scalingParameters[i] = scalingParameters[i-1];
+                    previousScalingParameters[i] = scalingParameters[i];
+                }
+                br.close();
```

Comment:
You can skip this `br.close()` step if you use the try-with-resources statement: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

For another example of this checkout https://github.com/datastax/cassandra/blob/ds-trunk/src/java/org/apache/cassandra/tools/CompactionLogAnalyzer.java#L353-L353